docs/plugins/define-plugin-settings.md

```diff
@@ -0,0 +1,66 @@
+---
+id: define-plugin-settings
+title: How To Define Plugin Settings
+sidebar_label: How To Define Plugin Settings
+slug: /plugins/define-plugin-settings
+---
+
+# How To Define Plugin Settings
+
+The `.amplicationrc.json` file lets you configure settings for your custom Amplication plugin.
+This file lives at the root of your plugin directory.
+
+It contains two properties at the root level:
+
+- `settings`: User-facing configuration.
+- `systemSettings` - Internal configuration.
+
+```json title=".amplicationrc.json"
+{
+  "settings": {
+    "host": "localhost",
+    "port": 27017,
+    "user": "admin",
+    "password": "admin",
+    "dbName": "my-db"
+  },
+  "systemSettings": {
+    "requireAuth": true
+  }
+}
+```
+
+### `settings`
+
+The `settings` property contains values that will appear in the Amplication UI when a user installs your plugin. This allows the user to customize your plugin's behavior.
+
+For example:
+
+```json
+"settings": {
+  "apiKey": "123456789",
+  "endpoint": "https://example.com/api"
+}
+```
+
+Define all user-facing configuration in the `settings` object.
```

Comment:
add explanation on how to access the settings from the plugin code

Reply:
Hey @yuval-hazaz @mulygottlieb 

I noticed that the MySQL plugin accesses its settings by using a custom [`getPluginSettings`](https://github.com/amplication/plugins/blob/1ea32952a67981ed6f7dc52565a22796d67f1f22/plugins/db-mysql/src/index.ts#L63) function defined in `utils.ts` by the plugin author. But, this function only considers the `settings` parameter passed in and not the `systemSettings`. 

Is having plugin authors explicitly create their own `utils.ts` with a `getPluginSettings` function still the recommended approach for plugins to handle settings?